### DISCUSSION TITLE
Can semantic colorization be applied to user-defined string literals?

### DISCUSSION THREAD

**Discussion by emptyVoid** [microsoft/vscode-cpptools#8057] [Q&A]
![image](https://user-images.githubusercontent.com/13639229/130899945-c97692f5-4919-4c47-a82a-853571a4e17c.png)
I like how special sequences are highlighted in string literals.
Is there a setting to apply such colorization to user-defined string literals?

**Comment by bobbrow** [comment]
@Colengms are these strings being colorized by the textmate grammar, or by our semantic tokens?  I actually didn't expect the first one to have the extra colors in it, though the request is to get those colors in all of the examples.

**Reply by Colengms** [reply]
Those colors are coming from VS Code and the TextMate grammar for C++.  The repo for that grammar is here: https://github.com/jeff-hykin/better-cpp-syntax

One way to tell if a token color is coming from VS Code TextMate grammar or the C/C++ extension's semantic tokens, is to run `Developer: Inspect Editor Tokens and Scopes`, then click on the colored token.  If it's related to semantic tokens, that dialog will refer to semantic tokens.  Otherwise, the token was colored by VS Code's based on TextMate grammar.
